File: backend/backend-java/src/main/java/com/cayena/backend/entities/Product.java
package com.cayena.backend.entities;

import com.cayena.backend.dtos.requesties.ProductRequest;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty("product_id")
    private Long productId;
    private String name;
    private Integer quantityInStock;
    private BigDecimal unitPrice;
    private LocalDateTime dateOfCreation = LocalDateTime.now();
    private LocalDateTime dateOfTheLastUpdate;

    @ManyToOne
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;

    public static Product converterRequest(ProductRequest request, Supplier supplier) {
        Product product = new Product();
        product.setProductId(request.getId());
        product.setName(request.getName());
        product.setQuantityInStock(request.getQuantityInStock());
        product.setUnitPrice(request.getUnitPrice());
        product.setSupplier(supplier);
        return product;
    }
}

File: backend/backend-kotlin/src/main/kotlin/com/cayena/backendkotlin/dominio/Product.kt
package com.cayena.backendkotlin.dominio

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal
import java.time.LocalDateTime
import javax.persistence.*


@Entity
data class Product(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty("product_id")
    val productId: Long? = null,
    var name: String,
    @JsonProperty("quantity_in_stock")
    var quantityInStock: Int,
    @JsonProperty("unit_price")
    var unitPrice: BigDecimal,
    @JsonProperty("date_of_creation")
    val dateOfCreation: LocalDateTime? = LocalDateTime.now(),
    @JsonProperty("date_of_the_last_update")
    var dateOfTheLastUpdate: LocalDateTime? = LocalDateTime.now(),

    @ManyToOne
    @JoinColumn(name = "supplier_id")
//    @JsonProperty("supplier_id")
    var supplier: Supplier
)
